.github/workflows/auto_cherry_pick.yml

```diff
@@ -0,0 +1,42 @@
+# CI stages to execute against all branches on PR merge
+name: auto_cherry_pick_commits
+
+on:
+  push
+
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+jobs:
+  branch-matrix:
+    name: Generate a branch matrix to apply cherrypicks
+    runs-on: ubuntu-latest
+    outputs:
+      branches: ${{ steps.set-matrix.outputs.branches }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - id: set-matrix
+        run: echo "::set-output name=branches::$(git branch -rl --sort=-authordate 'origin/6.*.z' --format='%(refname:lstrip=-1)' | head -n2 | jq -cnR '[inputs | select(length>0)]')"
+
+  auto_cherry_picking:
+    name: Auto Cherry picking to branches
+    needs: branch-matrix
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        to_branch: ${{ fromJson(needs.branch-matrix.outputs.branches) }}
+    # skipping PRs remote target_branch from cherrypicking again in self with the if condition
+    if: ${{ matrix.to_branch }} != github.base_ref
```

Comment:
I think we shouldn't auto-cherrypick 6.10.z or 6.9.z directly `on push` since some changes might not be required for 6.9.z or vice versa.

Instead, I would suggest cherrypicking based on included labels, for which I found the example herein https://github.com/carloscastrojumo/github-cherry-pick-action#example
(And when considering label-based triggers, we need to add some permissions for adding those labels)

@mshriver @JacobCallahan any thoughts?

Reply:
Read documents again and you are right @Gauravtalreja1 , we need that if condition ! Fixed it in latest commit !